editor/src/components/inspector/fill-hug-fixed-control.tsx

```diff
@@ -504,3 +711,110 @@ const anySelectedElementGroupOrChildOfGroup = createSelector(
     return selectedViews.some(elementOrAnyChildGroup)
   },
 )
+
+const allElementsAreGroupChildren = createSelector(
+  metadataSelector,
+  (store: MetadataSubstate) => store.editor.elementPathTree,
+  selectedViewsSelector,
+  (metadata, pathTrees, selectedViews): boolean => {
+    if (selectedViews.length === 0) {
+      return false
+    }
+    function elementOrAnyChildGroup(path: ElementPath) {
+      return treatElementAsGroupLike(metadata, EP.parentPath(path))
+    }
+    return selectedViews.every(elementOrAnyChildGroup)
+  },
+)
+
+function checkGroupChildConstraint(
+  dimension: LayoutPinnedProp,
+  selectedViews: ElementPath[],
+  allElementProps: AllElementProps,
+): GroupChildConstraintOptionType | 'mixed' {
+  const constrained = selectedViews.filter((path) =>
+    getSafeGroupChildConstraintsArray(allElementProps, path).includes(dimension),
+  ).length
+  return constrained === selectedViews.length
```

Comment:
Since this is already in a function, I feel like this would be clearer just using regular if..else statements.

Reply:
done in [`167c79a` (#4159)](https://github.com/concrete-utopia/utopia/pull/4159/commits/167c79af2bf4960d669631f31064c0ccbf6199de)